File: android/src/main/kotlin/com/example/scgateway_flutter_plugin/models/SmallcaseInfoDTO.kt
package com.example.scgateway_flutter_plugin.models

import com.google.gson.annotations.SerializedName

data class SmallcaseInfoDTO(

        @SerializedName("type")
        val type: String? = null,

        @SerializedName("name")
        val name: String? = null,

        @SerializedName("shortDescription")
        val shortDescription: String? = null,

        @SerializedName("publisherName")
        val publisherName: String? = null,

        @SerializedName("blogURL")
        val blogURL: String? = null,

        @SerializedName("nextUpdate")
        val nextUpdate: String? = null,

        @SerializedName("rebalanceSchedule")
        val rebalanceSchedule: String? = null,

        @SerializedName("lastRebalanced")
        val lastRebalanced: String? = null
)

File: example/lib/app/models/SmallcaseInfoDTO.dart

class SmallcaseInfoDTO {
  final String? type;
  final String? name;
  final String? shortDescription;
  final String? publisherName;
  final String? blogURL;
  final String? nextUpdate;
  final String? rebalanceSchedule;
  final String? lastRebalanced;

  SmallcaseInfoDTO({
    this.type,
    this.name,
    this.shortDescription,
    this.publisherName,
    this.blogURL,
    this.nextUpdate,
    this.rebalanceSchedule,
    this.lastRebalanced,
  });

  factory SmallcaseInfoDTO.fromJson(Map<String, dynamic> parsedJson) {
    return SmallcaseInfoDTO(
      type: parsedJson['type'],
      name: parsedJson['name'],
      shortDescription: parsedJson['shortDescription'],
      publisherName: parsedJson['publisherName'],
      blogURL: parsedJson['blogURL'],
      nextUpdate: parsedJson['nextUpdate'],
      rebalanceSchedule: parsedJson['rebalanceSchedule'],
      lastRebalanced: parsedJson['lastRebalanced'],
    );
  }
}
